gobblin-metrics/src/main/java/gobblin/metrics/graphite/GraphiteReporter.java

```diff
@@ -0,0 +1,296 @@
+/* (c) 2014 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.metrics.graphite;
+
+import java.io.IOException;
+import java.util.SortedMap;
+import java.util.concurrent.TimeUnit;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Histogram;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.Metered;
+import com.codahale.metrics.MetricFilter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Snapshot;
+import com.codahale.metrics.Timer;
+import com.codahale.metrics.graphite.GraphiteSender;
+
+import gobblin.metrics.ContextAwareScheduledReporter;
+import gobblin.metrics.Measurements;
+import gobblin.metrics.MetricContext;
+
+
+/**
+ * An implementation of {@link gobblin.metrics.ContextAwareScheduledReporter} that reports
+ * metrics to Graphite.
+ *
+ * @see <a href="http://graphite.wikidot.com/">Graphite - Scalable Realtime Graphing</a>.
+ *
+ * @author ynli
+ */
+public class GraphiteReporter extends ContextAwareScheduledReporter {
```

Comment:
We should agree on an API for generating the reporters (KafkaReporter and OutputStreamReporter use the codahale API). I agree with the advantage of this method of managing reporters from the MetricContext itself, but I also like to match existing APIs to reduce confusion. 
Unfortunately the two Builder APIs don't match easily while maintaining elegant, simple code, but maybe there's some ok way of doing it.